### PR TITLE
Adding new sequence diagrams with SF2.0 changes

### PR DESCRIPTION
**Pull request by subhankarc**
Check this for better viewing

https://github.com/subhankarc/service-fabrik-broker/blob/gh-pages/seqdiag.md

### CODE REVIEW

File: src/architecture/SF2.0-basics.seq [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Broker => APIServer [label = "Create resource with state as in_queue/update/delete", return = "true"];
+  }
+
+  APIServer -> Manager [label = "Trigger watcher for the request"]
+  Manager ->> Service [label = "service deploy/update/delete"]
+  Manager -> APIServer [label = "sets resource state as in_progress", return = "true"];
+
+  # Consumer asks for last operation status
+  Consumer => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-basics.seq]: remove director

File: src/architecture/SF2.0-basics.seq [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+  # Manager asks for last operation status
+  Manager => Service [label = "task state?", return = "succeeded"]
+    Manager => APIServer [label = "sets last operation as succeeded"]
+
+
+  # controller asks for last operation status
+  Consumer => Broker [label = "last operation?", return = "succeeded"] {
+    Broker => APIServer [label = "last operation state of director resource??", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-basics.seq]: remove director

File: src/architecture/SF2.0-service-binding-creation.seq [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  span_height = 0;   // default value is 40
+
+  # Change note color
+  default_note_color = lightblue;
+
+  # actors (in order left to right)
+  User; Controller; Broker; APIServer; BOSHManager; Agent; Service;
+
+  # trigger service binding creation
+  User => Controller [label = "cf create-service-   binding or -key", return = "credentials"] {
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-binding-creation.seq]: remove the cf notion, or use cf create-service-key

File: src/architecture/SF2.0-service-instance-bosh-backup-online.seq [hunk 4]
```diff
@@ -37,17 +34,11 @@ seqdiag {
     BackupManager => IaaS [label = "create backup record",return = "true"];
     APIServer <-- BackupManager [ label = "Update backup status as in progress" ];
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-backup-online.seq]: change it to sync req

File: src/architecture/SF2.0-basics.seq [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  # Consumer asks for last operation status
+  Consumer => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
+  }
+
+  Manager -> Manager [ label = "Start Poller" ];
+
+  # Manager asks for last operation status
+  Manager => Service [label = "task state?", return = "in progress"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-basics.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-basics.seq [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    Manager => APIServer [label = "sets last operation as in progress"]
+
+
+  # Manager asks for last operation status
+  Manager => Service [label = "task state?", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-basics.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    BOSHManager => APIServer [label = "deletes the director resource"]
+
+
+  # controller asks for last operation status
+  Controller => Broker [label = "last operation?", return = "succeeded"] {
+    Broker => APIServer [label = "last operation state of director resource??", return = "resource removed"]
+    Broker => APIServer [label = "remove the lock resource", return = "succeeded"]
+  }
+
+  User <<-- Controller [label = "creation completed\nno actual callback"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq]: deletion

File: src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  
+  # controller asks for last operation status
+  Controller => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
+  }
+
+  BOSHManager -> BOSHManager [ label = "Start Poller" ];
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    BOSHManager => APIServer [label = "sets last operation as in progress"]
+
+  # bosh starts undeployment
+  Bosh => Service [label = "monit stop all"]
+  # bosh continues undeployment
+  Bosh => IaaS [label = "release resources"]
+  Broker <<-- Bosh [label = "deletion completed\nno actual callback"]
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
+    BOSHManager => APIServer [label = "sets last operation as in progress"]
+
+  # bosh starts undeployment
+  Bosh => Service [label = "monit stop all"]
+  # bosh continues undeployment
+  Bosh => IaaS [label = "release resources"]
+  Broker <<-- Bosh [label = "deletion completed\nno actual callback"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-deprovisioning.seq]: should go to BOSHManager, not broker

File: src/architecture/SF2.0-service-instance-bosh-provisioning.seq [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+  # controller asks for last operation status
+  Controller => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
+  }
+
+  BOSHManager -> BOSHManager [ label = "Start Poller" ];
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-provisioning.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-provisioning.seq [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    BOSHManager => APIServer [label = "sets last operation as in progress"]
+
+  # bosh continues deployment
+  Bosh => Service [label = "monit start all"]
+  BOSHManager <<-- Bosh [label = "deployment completed\nno actual callback"]
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-provisioning.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-update-automated.seq [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
+  BOSHManager => APIServer [label = "sets last operation as in progress"]
+
+  # bosh continues deployment
+  Bosh => Service [label = "monit start all"]
+  BOSHManager <<-- Bosh [label = "deployment completed\nno actual callback"]
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-update-automated.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-update-automated.seq [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+
+  # controller asks for last operation status
+  Controller => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
+  }
+
+  BOSHManager -> BOSHManager [ label = "Start Poller" ];
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-update-automated.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-update-automated.seq [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+
+  # Change note color
+  default_note_color = lightblue;
+
+  # actors (in order left to right)
+  Admin; Controller; Broker; APIServer; BOSHManager; Bosh; Service; IaaS;
+
+  # trigger update of outdated service instances
+  Admin ->> Broker [label = "trigger update of outdated deployments"] {
+    Broker => Bosh [label = "download manifest(s)"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-update-automated.seq]: Ideally Broker should not contact BOSH

File: src/architecture/SF2.0-service-instance-bosh-update.seq [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+  # controller asks for last operation status
+  Controller => Broker [label = "last operation?", return = "in progress"] {
+    Broker => APIServer [label = "last operation of director resource?", return = "processing"]
+  }
+
+  BOSHManager -> BOSHManager [ label = "Start Poller" ];
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "in progress"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-update.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

File: src/architecture/SF2.0-service-instance-bosh-update.seq [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+  BOSHManager => APIServer [label = "sets last operation as in progress"]
+
+  # bosh continues deployment
+  Bosh => Service [label = "monit start all"]
+  BOSHManager <<-- Bosh [label = "deployment completed\nno actual callback"]
+
+  # BOSHManager asks for last operation status
+  BOSHManager => Bosh [label = "task state?", return = "succeeded"]
```
Comment (subhankarc) [inline src/architecture/SF2.0-service-instance-bosh-update.seq]: Use like this.

```
    FabrikApi => FabrikApi [ label = "check status" ]{	
      FabrikApi => APIServer [label = "backup resource state?", return = "in_queue"];	
    }
```

**Comment by subhankarc** [comment]
@amitmalav @jagadish-kb @kamath-prasad Please review this. Once merged, I'll update the Wiki. For better viewing, please use the MD file I have provided.

File: src/architecture/SF2.0-basics.seq [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+
+  # Change note color
+  default_note_color = lightblue;
+
+  # actors (in order left to right)
+  Consumer; Broker; APIServer; Manager; Service;
+
+  # trigger provisioning of service instance
+  Consumer => Broker [label = "Create/Update/Delete", return = "ok"] {
+    Broker => APIServer [label = "Create resource with state as in_queue/update/delete", return = "true"];
```
Comment (kamath-prasad) [inline src/architecture/SF2.0-basics.seq]: state will just be `in_queue` , correct ?

File: src/architecture/SF2.0-service-instance-bosh-provisioning.seq [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+  User ->> Controller [label = "cf create-service"] {
+    Controller => Broker [label = "run provisioning", return = "ok"] {
+      Broker => APIServer [label = "create lock resource on deployment", return = "true"];
+      Broker => APIServer [label = "Create a director resource with state as in_queue", return = "true"];
+    }
+  }
+
+  APIServer -> BOSHManager [label = "Trigger watcher for new create request"]
+  BOSHManager ->> Bosh [label = "bosh deploy"]
+  BOSHManager -> APIServer [label = "sets resource state as in_progress", return = "true"];
```
Comment (kamath-prasad) [inline src/architecture/SF2.0-service-instance-bosh-provisioning.seq]: maybe you can mention `director resource` here.

File: src/architecture/SF2.0-basics.seq (hunk 18 above)
Comment (subhankarc) [reply src/architecture/SF2.0-basics.seq]: This diagram presents all the CRUD operations.
For create, the state is `in_queue`, for update, it is `update` and for delete, it is `delete`

File: src/architecture/SF2.0-service-instance-bosh-provisioning.seq (hunk 19 above)
Comment (subhankarc) [reply src/architecture/SF2.0-service-instance-bosh-provisioning.seq]: cool!

**Pull request merged**